### DISCUSSION TITLE
Sending build status to another application

### DISCUSSION THREAD

**Discussion by gwaribe** [gocd/gocd#12730] [Q&A]
Hello,
I want to send the build status of a pipeline to another application.
How can I use a webhook in this case, or what's the best way achieve that?

**Comment by chadlwilson** [comment]
There's some discussion about this here: https://groups.google.com/g/go-cd/c/Wjwn06iKv-0 with respect to Mattermost (Slack-like).

It depends on what the target application is. However, keep in mind that there is no such thing as a "generic" webhook format that would allow GoCD to communicate with another application. One of GoCD or the "other" application would need to define the format/API of the webhook; and GoCD doesn't have a "default" format for a webhook format.

If you are thinking of writing a custom application, you could make it conform to one of the formats used by an existing plugin at https://www.gocd.org/plugins/#notification e.g https://github.com/matic-insurance/gocd-http-notifications-plugin (third party plugin, not endorsed) or https://github.com/ashwanthkumar/gocd-slack-build-notifier if you want to conform to Slack's format (also third party plugin, not endorsed) 

Alternatively, you can write your own "notification" plugin for GoCD: https://plugin-api.gocd.org/current/notifications/ which would allow you to use whatever format you like.

**Reply by gwaribe** [reply]
Thank you for this guide.
